tls/s2n_cipher_suites.c

```diff
@@ -995,6 +1006,7 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t * wire,
     /* Settle for a cipher with a higher required proto version, if it was set */
     if (higher_vers_match) {
         conn->secure.cipher_suite = higher_vers_match;
+        conn->handshake_params.our_chain_and_key = s2n_get_compatible_cert_chain_and_key(conn, higher_vers_match);
```

Comment:
Is this necessary? Trying to follow the logic, but it looks like `our_chain_and_key` is set with `match` and then `higher_vers_match` is set to `match` so wouldn't `our_chain_and_key` already have a cert set for `higher_vers_match`?

Reply:
the cert set might not match the one that was compatible with `higher_vers_match` since the loop continues to and conn->handshake_params.our_chain_and_key may be overwritten with a compatible cert for future "potential" matches.

I think it's simpler if I just add state for `higher_vers_match_cert`